internal_ws/ykcompile/src/arch/x86_64/mod.rs

```diff
@@ -1286,7 +1286,11 @@ impl TraceCompiler {
     }
 
     /// Emit a return instruction.
-    fn ret(&mut self, gl: Vec<(&Guard, HashMap<&Local, Location>, DynamicLabel)>) {
+    fn ret(
+        &mut self,
+        gl: Vec<(&Guard, HashMap<&Local, Location>, DynamicLabel)>,
+        stitch_trace: bool,
```

Comment:
This function is already very overloaded AFAICS in that it does more than just "emit a return instruction". I'm nervous about bloating it further with `stitch_trace` especially as I'm not sure why it's a bool -- can you clarify why it might be true/false?

Reply:
This is a restriction of the current API (where API is used very loosely), which we'll likely change in the future. The user needs to return `true/false` in `interp_step`, where `false` means loop forever. But currently nothing keeps them from only return true, e.g.:

```
fn interp_step() -> bool {
    // do interpreter stuff
    true
}
```

Without a condition that returns `false` in one case, we would loop forever with no way of exiting the trace. So in such a case we do not want to stitch the trace. Although, similarly if the function only returns `false` we would also loop forever, so maybe we do require some extra check. My initial plan was to do this when we turn this into a proper API in a future PR.

Reply:
I thought we weren't going to offer any solution for "I don't want stitching", because we couldn't think of a useful use case where someone might reasonably want to do that?

Reply:
[We've discussed splitting `ret` a week or so ago. I think we need this more now than ever]